### PR TITLE
chores(linter): enable rest of the linters

### PR DESCRIPTION
**Pull request by mangalaman93**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -5,17 +5,21 @@ run:
 linters-settings:
   lll:
     line-length: 120
+  staticcheck:
+    checks:
+      - all
+      - '-SA1019' # disable the rule
```
Comment (anurags92) [inline .golangci.yml:11]: Please mention what this rule checks and why we want to disable it?

File: manifest_test.go [hunk 2]
```diff
@@ -158,58 +144,6 @@ func buildTable(t *testing.T, keyValues [][]string, bopts table.Options) *table.
 	return tbl
 }
 
-func TestOverlappingKeyRangeError(t *testing.T) {
-	dir, err := ioutil.TempDir("", "badger-test")
```
Comment (joshua-goldstein) [inline manifest_test.go:162]: Perhaps instead of deleting we can skip this test?

File: table/table_test.go [hunk 3]
```diff
@@ -387,9 +387,9 @@ func TestConcatIterator(t *testing.T) {
 	tbl := buildTestTable(t, "keya", 10000, opts)
 	tbl2 := buildTestTable(t, "keyb", 10000, opts)
 	tbl3 := buildTestTable(t, "keyc", 10000, opts)
-	defer tbl.DecrRef()
-	defer tbl2.DecrRef()
-	defer tbl3.DecrRef()
+	defer func() { require.NoError(t, tbl.DecrRef()) }()
+	defer func() { require.NoError(t, tbl2.DecrRef()) }()
+	defer func() { require.NoError(t, tbl2.DecrRef()) }()
```
Comment (joshua-goldstein) [inline table/table_test.go:392]: tb13

File: value.go [hunk 4]
```diff
@@ -820,7 +819,9 @@ func (vlog *valueLog) write(reqs []*request) error {
 		endOffset := atomic.AddUint32(&vlog.writableLogOffset, n)
 		// Increase the file size if we cannot accommodate this entry.
 		if int(endOffset) >= len(curlf.Data) {
```
Comment (joshua-goldstein) [inline value.go:821]: do we need `>=` here?

**Review by joshua-goldstein — CHANGES_REQUESTED** [review]
see comments

File: manifest_test.go (hunk 2 above)
Comment (mangalaman93) [reply manifest_test.go:162]: done

File: table/table_test.go (hunk 3 above)
Comment (mangalaman93) [reply table/table_test.go:392]: done

File: value.go (hunk 4 above)
Comment (mangalaman93) [reply value.go:821]: Added a comment for now

**Review by joshua-goldstein — APPROVED** [review]

**Pull request merged**